8295944: Move the Http2TestServer and related classes into a package of its own

jaikiran opened the pull request:
Can I please get a review of this change which addresses https://bugs.openjdk.org/browse/JDK-8295944?

As noted in the issue, the HttpClient testing helper infrastructure code resides in a default (unnamed) packages within the test hierarchy. The commit in this PR moves those classes into a specific package. The tests have been updated to refer to them correctly with proper package name references.
Additionally as noted in the issue, the `@modules` declaration used in the httpclient tests, were all the same and were repeated in all these tests. The commit in this  PR moves those module references into the `TEST.properties` to centralize such reference. Any test which needs additional modules can then use specific `@modules` declaration in the test definition.

No functional changes have been done in this PR.

tier1, tier2 and tier3 testing passed after these changes.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295944](https://bugs.openjdk.org/browse/JDK-8295944): Move the Http2TestServer and related classes into a package of its own


### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12160/head:pull/12160` \
`$ git checkout pull/12160`

Update a local copy of the PR: \
`$ git checkout pull/12160` \
`$ git pull https://git.openjdk.org/jdk pull/12160/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12160`

View PR using the GUI difftool: \
`$ git pr show -t 12160`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12160.diff">https://git.openjdk.org/jdk/pull/12160.diff</a>

</details>

dfuch commented on test/jdk/java/net/httpclient/FlowAdapterSubscriberTest.java:
```diff
@@ -62,15 +66,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/ExecutorShutdown.java:
```diff
@@ -26,15 +26,11 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/CookieHeaderTest.java:
```diff
@@ -25,15 +25,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/HeadTest.java:
```diff
@@ -25,15 +25,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/PathSubscriber/BodySubscriberOfFileTest.java:
```diff
@@ -27,18 +27,16 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/RedirectWithCookie.java:
```diff
@@ -24,15 +24,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/Response1xxTest.java:
```diff
@@ -49,14 +50,11 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/RetryWithCookie.java:
```diff
@@ -25,15 +25,12 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/StreamingBody.java:
```diff
@@ -25,15 +25,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/UnauthorizedTest.java:
```diff
@@ -29,15 +29,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/UserCookieTest.java:
```diff
@@ -26,15 +26,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/BasicRedirectTest.java:
```diff
@@ -24,15 +24,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/FilePublisher/FilePublisherPermsTest.java:
```diff
@@ -29,14 +29,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/FilePublisher/FilePublisherTest.java:
```diff
@@ -26,14 +26,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/MaxStreams.java:
```diff
@@ -26,15 +26,10 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/PathSubscriber/BodyHandlerOfFileDownloadTest.java:
```diff
@@ -26,18 +26,18 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          jdk.httpserver
```
```suggestion
```

dfuch commented on test/jdk/java/net/httpclient/PathSubscriber/BodyHandlerOfFileTest.java:
```diff
@@ -27,18 +27,17 @@
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          jdk.httpserver
```
```suggestion
```

dfuch reviewed (changes requested):
Very nice refactoring overall. Thanks for taking that on! There are a few places where some lines should have been removed but were not. I made some suggestions. Also there are 9 classes that have kept their custom `@modules` clause because they want to access 
```
 *          java.base/sun.net.www
 *          java.base/sun.net 
 ```
 I wonder if we should just add these two in TEST.properties - and then remove the `@modules` from these nine classes as well?

jaikiran commented:
Hello Daniel,

> There are a few places where some lines should have been removed but were not. I made some suggestions.

Thank you for catching these. These were genuine issues and were a leftover from my `@modules` removal. I've fixed these now in the updated PR. I went through all the files again and did a search (with a different search term this time) and I think have now covered all these leftovers.

jaikiran commented:
> Also there are 9 classes that have kept their custom @modules clause because they want to access
>
>           java.base/sun.net.www
>           java.base/sun.net 
>
> I wonder if we should just add these two in TEST.properties - and then remove the @modules from these nine classes as well?

I think that's reasonable. When I changed most of these tests to remove `@modules` from their test definition, my selection of "common" declarations missed out these two declarations. I've now updated the `TEST.properties` to include these and also updated the relevant tests to remove the `@modules` declaration.

After this change, there remains few tests which use only a limited subset of the modules delcared in `TEST.properties`. I have left those tests as-is (for example `CancelledResponse.java`), since it didn't look necessary to remove those `@modules`. Let me know if you prefer otherwise and I'll update them as well.

Local testing with the latest state of the PR passed. I've now triggered a broader testing with these changes.

jaikiran commented:
Thank you Daniel for the review. tier1, tier2, tier3 testing came back successful.

jaikiran commented:
/integrate
